### DISCUSSION TITLE
Configure-time library dependency problem

### DISCUSSION THREAD

**Discussion by undisputed-seraphim** [zephyrproject-rtos/zephyr#47726] [Q&A]
There's no question label so I went with bug.

I'm attempting to build a library on Zephyr with West, called Foo. This library depends on Bar to be built and installed to some prefix before its CMakeLists will even pass the configure stage. Unfortunately I own neither Foo nor Bar so I'm not in a position to fix either library.

Compiling Bar and manually moving libbar.a to the toolchain install prefix allows Foo to be configured and built successfully, so it really is just a matter of adding an install step. However, I've experimented with several ways with West to try to get it to produce and install Bar, but to no success. I've tried adding Foo and Bar as modules and setting Bar as a dependency of Foo in `module.yml`, no luck. It appears that in every scenario (modules, external libs, etc), West tries to configure everything in 1 step, and then build+link everything in another step, but I need some way to say "Build and install Bar to a prefix, then get Foo to look for Bar at this prefix".

How can I achieve this?

**Comment by yashi** [comment]
You don't usually _install_ a library when you are working in the Zephyr build system.  You just build your library along with its dependencies in your [workspace](https://docs.zephyrproject.org/latest/develop/west/workspaces.html), link them all together to make a final binary file.

I assume that your libraries, Foo and Bar, are for Linux or some other general purpose OSes? Or, are they for a RTOS which assumes dependencies are installed on a host OS?  For either cases, you must modify the build system of Foo and Bar and must integrate them to Zephyr's build system.
